src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,183 @@
+            if (input.length == 2) {
+                try {
+                    this.type = "mark";
+                    list.getTaskList().get(Integer.parseInt(input[1]) - 1).mark();
+                    System.out.println("----------------------\n" + "Congrats on completing :)\n" +
+                            listOfActions.get(Integer.parseInt(input[1]) - 1) + "\n----------------------\n");
+                    this.isErreneous = false;
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("");
+                }
+            }
+        } else if (input[0].equals("unmark")) {
+            if (input.length == 2) {
+                try {
+                    this.type = "unmark";
+                    list.getTaskList().get(Integer.parseInt(input[1]) - 1).unMark();
+                    System.out.println("----------------------\n" + "One more mission ;)\n" +
+                            listOfActions.get(Integer.parseInt(input[1]) - 1) + "\n----------------------\n");
+                    this.isErreneous = false;
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException(" ");
+                }
+            }
+
+        } else if (input[0].equals("delete")) {
+            if (input.length == 2) {
+                try {
+                    int pos = Integer.parseInt(input[1]) - 1;
+                    this.type = "delete";
+                    list.delete(pos);
+                    this.isErreneous = false;
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DukeException("");
+                }
+            }
+        } else if (input[0].equals("find")) {
+            String out = findTasks(input[1]);
+            this.type = "find";
+            this.isErreneous = false;
+            if (out.isBlank()) {
+                System.out.println("----------------------\n" + "Oops nothing fits the description :(\n"
+                        + "----------------------\n");
+            } else {
+                System.out.println("----------------------\n" + out + "----------------------\n");
+            }
+        }
+
+
+        if (this.type.equals("error")) {
+            this.isErreneous = true;
+        }
+    }
+
+    /**
+     * Gets the type of userInput.
+     * @return The String representation of the type of userInput.
+     */
+
+    public String getType() {
+        return this.type;
+    }
+
+    /**
+     * Gets the Boolean value of whether the input is valid.
+     * @return The Boolean value of the input if valid.
+     */
+    public Boolean isErreneous() {
+        return this.isErreneous;
+    }
+
+    /**
+     * Gets the Boolean value if userInput is a Task.
+     * @return The Boolean value if userInput is a Task.
+     */
+    public Boolean getIsAction() {
+        return this.isAction;
+    }
+
+    private void bye(ArrayList<Task> listOfActions, File file) {
```

Comment:
perhaps can give a more descriptive name to this method instead of bye (eg. a name related to what the method is doing)